### ISSUE TITLE
javascript heap size error while deploying.

### ISSUE THREAD

**Issue by teajoon** [danielcondemarin/serverless-next.js#457]
There is an error to deploy in new versions. - 1.14.0, 1.15.0-alpha.0
I used 1.12.0-alpha.6 version. And i checked the new version in release tab. 
But it makes error like bottom unlike 1.12.0-alpha.6
**I enlarged the heap size as much as i can(12gb). However there was same error.**
I don't know what is different in 1.14.0, 1.15.0-alpha.0. But I hope it will work well in new version.

```
Error: Command was killed with SIGABRT (Aborted): node_modules/.bin/next build
FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
 1: 0x100069ede node::Abort() [/usr/local/bin/node]
 2: 0x10006a62e node::errors::TryCatchScope::~TryCatchScope() [/usr/local/bin/node]
 3: 0x1001b5827 v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [/usr/local/bin/node]
 4: 0x1001b57c4 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [/usr/local/bin/node]
 5: 0x1005bc702 v8::internal::Heap::FatalProcessOutOfMemory(char const*) [/usr/local/bin/node]
 6: 0x1005bec33 v8::internal::Heap::CheckIneffectiveMarkCompact(unsigned long, double) [/usr/local/bin/node]
 7: 0x1005bb168 v8::internal::Heap::PerformGarbageCollection(v8::internal::GarbageCollector, v8::GCCallbackFlags) [/usr/local/bin/node]
 8: 0x1005b9325 v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [/usr/local/bin/node]
 9: 0x1005c5bf8 v8::internal::Heap::AllocateRawWithLightRetry(int, v8::internal::AllocationSpace, v8::internal::AllocationAlignment) [/usr/local/bin/node]
10: 0x1005c5c4f v8::internal::Heap::AllocateRawWithRetryOrFail(int, v8::internal::AllocationSpace, v8::internal::AllocationAlignment) [/usr/local/bin/node]
11: 0x100597ad4 v8::internal::Factory::NewRawTwoByteString(int, v8::internal::PretenureFlag) [/usr/local/bin/node]
12: 0x1006de96a v8::internal::String::SlowFlatten(v8::internal::Isolate*, v8::internal::Handle<v8::internal::ConsString>, v8::internal::PretenureFlag) [/usr/local/bin/node]
13: 0x1001d2509 v8::String::Utf8Length(v8::Isolate*) const [/usr/local/bin/node]
14: 0x10004f0b0 node::Buffer::(anonymous namespace)::ByteLengthUtf8(v8::FunctionCallbackInfo<v8::Value> const&) [/usr/local/bin/node]
15: 0x107926ad2e41 
info  - Loaded env from /Users/teajoon/Documents/yellowLiker/popupcinema-seeso-front/.env
Creating an optimized production build...

<--- Last few GCs --->

[32180:0x102841000]    43881 ms: Mark-sweep 1311.3 (1384.9) -> 1311.3 (1370.9) MB, 288.2 / 0.0 ms  (average mu = 0.150, current mu = 0.000) allocation failure GC in old space requested
[32180:0x102841000]    44254 ms: Mark-sweep 1342.9 (1401.9) -> 1342.3 (1390.9) MB, 266.1 / 0.1 ms  (average mu = 0.223, current mu = 0.288) allocation failure GC in old space requested


<--- JS stacktrace --->

==== JS stack trace =========================================

    0: ExitFrame [pc: 0x107926ad2e41]
Security context: 0x161da279d9d1 <JSObject>
    1: byteLength(aka byteLength) [0x161deb55d8d1] [buffer.js:~499] [pc=0x107928174025](this=0x161d951025b1 <undefined>,0x161db0c87799 <Very long string[15673240]>,0x161dfd2a46f9 <String[4]: utf8>)
    2: arguments adaptor frame: 3->2
    3: fromString(aka fromString) [0x161deb56ef69] [buffer.js:~325] [pc=0x107927529874](this=0x161d951025b1 <undefined>,0x161...
```

**Comment by lone-cloud** [comment]
I would first try the latest versions. How did you increase the heap size? Did you try passing `max-old-space-size=4096` arg to the build via `build.args`?

**Comment by teajoon** [comment]
> I would first try the latest versions. How did you increase the heap size? Did you try passing `max-old-space-size=4096` arg to the build via `build.args`?

i tried like this.
```
node --max-old-space-size=2048 node_modules/serverless/bin/serverless.js --component=componentname --debug deploy
```

**Comment by lone-cloud** [comment]
I don't think that would work as `next build` would be forked as a new process and it wouldn't have the increased memory size. I haven't tried it, but the solution might be to provide the `build.args` input as `['build', '--max-old-space-size=4096']` in the serverless.yml

**Comment by teajoon** [comment]
> I don't think that would work as `next build` would be forked as a new process and it wouldn't have the increased memory size. I haven't tried it, but the solution might be to provide the `build.args` input as `['build', '--max-old-space-size=4096']` in the serverless.yml

Thanks for guide. However It also makes same error. 
My inputs is like this.

```
inputs:
    name:
      defaultLambda: lambdaName
      apiLambda: lambdaApiName
    memory:
      defaultLambda: 512
      apiLambda: 1024
    bucketName: bucketname
    build.args: ['build', '--max-old-space-size=8192']
    cloudfront: 
      api/*:
        forward:
          headers:
            [
              Authorization,
              Content-Type
            ]
```

**Comment by lone-cloud** [comment]
I'm not sure if that way works. How about:
```
inputs:
    name:
      defaultLambda: lambdaName
      apiLambda: lambdaApiName
    memory:
      defaultLambda: 512
      apiLambda: 1024
    bucketName: bucketname
    build:
        args: ['build', '--max-old-space-size=8192']
    cloudfront: 
      api/*:
        forward:
          headers:
            [
              Authorization,
              Content-Type
            ]
```

**Comment by teajoon** [comment]
@lone-cloud

Thanks for reply.
I put the code u guided, and it makes another error, like this.
```
pages-manifest not found. Check if `next.config.js` target is set to 'serverless'
```

I founded this error message in other issue #447 
The isssue is closed with version 1.14.0. However, I think it is still going on.

My current serverless.yml file is like this.
```
componentname:
  component: serverless-next.js@1.14.0
  inputs:
    name:
      defaultLambda: lambdaname
      apiLambda: lambdaapiname
    memory:
      defaultLambda: 512
      apiLambda: 1024
    bucketName: bucketname
    build:
      args: ['build', '--max-old-space-size=8192']
    cloudfront: 
      api/*:
        forward:
          headers:
            [
              Authorization,
              Content-Type
            ]
```

**Comment by teajoon** [comment]
It is close since it is solved with build option(--max-old-space-size=8192),
But it makes another error, so i made a new issue.
